solver/src/naive_solver/multi_order_solver.rs

```diff
@@ -0,0 +1,369 @@
+#![allow(dead_code)]
+
+use super::single_pair_settlement::{AmmSwapExactTokensForTokens, SinglePairSettlement};
+use crate::settlement::Trade;
+use anyhow::{anyhow, Result};
+use model::order::{OrderCreation, OrderKind};
+use num::{bigint::Sign, BigInt};
```

Comment:
There is a BigUInt type which is probably more appropriate in most places.

Reply:
The reason I used BigInt internally (rather than e.g. U512 which would probably be sufficient width wise), is that some of the arithmetic can underflow (e.g. when computing excess.sell_volume - excess.buy_volume in compute_uniswap_out).

Do you think it's better to have the internal structs use the more strict type and only convert in the methods directly?

Reply:
If that underflow is temporary in that function and it always receives and returns positive numbers then I would use the BigInteger only inside there and keep the interface clean.